### PR TITLE
Gateway node 3887

### PR DESCRIPTION
**Pull request by cong-or**
As we gradually start to allow full public node participation in the network, we will start by deploying gateway nodes as an entry point. Gateway nodes can only consume from the internal network, they cannot _propagate/gossip/publish_.

This is a _temporary_ solution to prevent flooding attacks etc.

Below is a _crude_ diagram to illustrate the point.

A gateway node that is conditionally compiled (**_d_** in the diagram), 

**d** will be open to the wild - **A, B, C** will not be. 

Screenshot 2022-11-15 at 13 52 08

### CODE REVIEW

**Comment by cong-or** [comment]
@dkijania do these tests make sense?

File: src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/connections.rs [hunk 1]
```diff
@@ -108,6 +115,91 @@ pub fn node_trust_itself() {
         .unwrap();
 }
 
+#[ignore]
+#[test]
+fn gateway_gossip() {
```
Comment (dkijania) [inline src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/connections.rs:120]: hm, i don't think `cross_version` module is not a good place for this test method. Maybe it should be put into more meaningful module? like `gateway`?

File: src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/connections.rs [hunk 2]
```diff
@@ -108,6 +115,91 @@ pub fn node_trust_itself() {
...
+            SpawnParams::new(SERVER)
+                .gossip_interval(Duration::new(SERVER_GOSSIP_INTERVAL_SECS, 0))
+                .log_level(LogLevel::TRACE),
+        )
+        .unwrap();
+
+    //
+    // spin up gateway node
+    //
+    let mut gateway_node_binary = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
```
Comment (dkijania) [inline src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/connections.rs:158]: i wonder if `gateway` app is not available on path when running this test?

File: src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/stats.rs [hunk 3]
```diff
@@ -159,6 +163,66 @@ macro_rules! build_network {
...
+        100.into(),
+    ) {
+        Ok(_) => panic!("gateway node should not be able to send fragments"),
+        Err(err) => assert_eq!(
+            err.to_string(),
+            "Too many attempts failed (1) while trying to send fragment to node: ".to_string()
+        ),
+    };
+}
+
```
Comment (dkijania) [inline src/jormungandr/testing/jormungandr-integration-tests/src/networking/p2p/stats.rs:225]: i would add one more test just to be sure that gateway node function correctly with trusted nodes. For example we are configure gateway node not to publish but we need to make sure that it still sync with rest of the network correctly

**Comment by dkijania** [comment]
@cong-or looking good! i see one more potential test

**Pull request closed without merging**